arch/risc-v/src/common/riscv_swint.c

```diff
@@ -179,6 +187,31 @@ int riscv_swint(int irq, void *context, void *arg)
   riscv_registerdump(regs);
 #endif
 
+  /* Handle the syscall */
+
+  regs = riscv_handle_syscall(regs);
+
+  /* Report what happened.  That might difficult in the case of a context
+   * switch
+   */
+
+#ifdef CONFIG_DEBUG_SYSCALL_INFO
```

Comment:
should we move the log at the begin and end to riscv_handle_syscall?

Reply:
This is how it was and it makes sense to me.

Also, CURRENT_REGS is the interrupt context AFAIK, so this is the right place for this

Reply:
Regardless where the syscall come from(U-mode or S-mode), it will surprise the user who turn on the log, but the half of log(S-mode) is skipped.

Reply:
Hmm, logs will be produced for all user syscalls, but obviously not for the context handling ones. That is the only difference.

Reply:
Yes, but the S-mode context switch is important as U-mode context switch, so it's better to log both if user enable it.

Reply:
Done